Source/Graphic/HCPath.c

```diff
@@ -230,21 +205,362 @@ HCPathElement HCPathElementAt(HCPathRef self, HCInteger elementIndex) {
+        
+        // Compare the line segments of the polyline against the other path
+        for (HCInteger pointIndex = 1; pointIndex < polylinePointCount; pointIndex++) {
+            // Determine the end point of the line segment from the polyline
+            HCPoint p1 = HCPathPolylinePointAt(self, elementIndex, pointIndex);
+            HCInteger otherElementCount = HCPathElementCount(other);
+            for (HCInteger otherElementIndex = 0; otherElementIndex < otherElementCount; otherElementIndex++) {
+                // Extract the initial point of the other path polyline and use it as the start point of it's first line segment, or skip it if it is empty
+                HCInteger otherPolylinePointCount = HCPathPolylinePointCount(other, otherElementIndex);
+                if (otherPolylinePointCount <= 0) {
+                    continue;
+                }
+                HCPoint q0 = HCPathPolylinePointAt(other, otherElementIndex, 0);
+                
+                // Compare the line segments of the other polyline against the current line segment in the path
+                for (HCInteger otherPointIndex = 0; otherPointIndex < otherPolylinePointCount; otherPointIndex++) {
+                    // Determine the end point of the line segment from the other polyline
+                    HCPoint q1 = HCPathPolylinePointAt(other, otherElementIndex, otherPointIndex);
+                    
+                    // Find the intersection parameters for the lines formed by the line segment points
+                    HCReal t = 0.0;
+                    HCReal u = 0.0;
+                    HCContourLineLineIntersection(p0, p1, q0, q1, &t, &u);
+                    
+                    // Determine if they intersect within the bounds of the segments
+                    HCBoolean segmentsIntersect = t >= 0.0 && t <= 1.0 && u >= 0.0 && u <= 1.0;
+                    if (segmentsIntersect) {
+                        // Calculate the intersection point and call the intersection function
+                        HCReal tc = 1.0 - t;
+                        HCPoint point = HCPointMake(tc * p0.x + t * p1.x, tc * p0.y + t * p1.y);
+                        HCBoolean stopSearching = false;
+                        intersection(context, &stopSearching, self, other, point);
+                        if (stopSearching) {
+                            return;
+                        }
+                    }
+                    
+                    // Use the end point of the other polyline line segment as the start point of the next
+                    q0 = q1;
+                }
+            }
+            
+            // Use the end point of the polyline line segment as the start point of the next
+            p0 = p1;
+        }
+    }
+}
+
+void HCPathIntersects(void* context, HCBoolean* stopSearching, HCPathRef path, HCPathRef otherPath, HCPoint point) {
+    (void)path; // unused
+    (void)otherPath; // unused
+    (void)point; // unused
```

Comment:
Do we need these?

Reply:
Yes, otherwise compiling with -Wall gives a warning.